src/NEPTypes.jl

```diff
@@ -804,12 +828,16 @@ julia> compute_Mder(nep,3.0)[1:2,1:2]
         ## Sets the left and right projected basis and computes
         ## the underlying projected NEP
         m = size(nep.orgnep_Av,1);
-        B = Array{Array{eltype(W),2}}(undef, m);
+        T=eltype(eltype(nep.projnep_B_mem));
+
+        T_sub = SubArray{T,2,Array{T,2},Tuple{UnitRange{Int64},UnitRange{Int64}},false}
```

Comment:
Maybe `Matrix{T}` instead of `Array{T,2}`?

Reply:
Or perhaps: Maybe create the first B[i] and determine the type based on that.

Reply:
Ok.